Sources/BrowserServicesKit/ContentBlocking/UserScripts/surrogates.js

```diff
@@ -508,13 +506,14 @@
             if (element && element.onerror) {
                 element.onerror = () => {}
             }
-            if (!loadedSurrogates[result.matchedRule.surrogate]) {
+            try {
                 loadSurrogate(result.matchedRule.surrogate)
-                loadedSurrogates[result.matchedRule.surrogate] = true
-                // Trigger a load event on the original element
-                if (element && element.onload) {
-                    element.onload(new Event('load'))
-                }
+            } catch (e) {
+                duckduckgoDebugMessaging.log(`error loading surrogate: ${e.toString()}`)
+            }
+            // Trigger a load event on the original element
+            if (element && element.onload) {
```

Comment:
Probably could go either way, but should this be within the try block?

The rationale is that the `loadSurrogate` should always load on our side.

- The only way I can see it throwing is when there's an error state from loading multiple times let's say the global it declares is already declared etc.
- In these edge cases the site might end up depending on the load not firing.